src/main/java/agregator/filters/NewsFilter.java

```diff
@@ -0,0 +1,52 @@
+package agregator.filters;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import agregator.core.StoragesKeeper;
+import agregator.io.SettingsStorage;
+import agregator.structure.NewsItem;
+import agregator.structure.NewsPack;
+import agregator.utils.NewsProcessor;
+
+public class NewsFilter extends HandlerInterceptorAdapter {
+    private SettingsStorage settingsStorage;
+    
+    @Autowired
+    public NewsFilter(ServletContext context) {
+        settingsStorage = StoragesKeeper.getSettingsStorage(context);
+    }
+    
+    @Override
+    public void postHandle(HttpServletRequest request,
+                           HttpServletResponse responce,
+                           Object handler,
+                           ModelAndView modelAndView) throws Exception {
+        String category;
+        
+        if (settingsStorage.getCategoryFilterEnabled()) {
```

Comment:
is it possible to create delegate where logic of filtering is implemented, and this logic doesn't related to filtering mechanism?